### ISSUE TITLE
CORDEX-CMIP6 cmor tables

### ISSUE THREAD

**Issue by larsbuntemeyer** [PCMDI/cmor#677]
Hey everybody,
i have some issues since i am working on CMOR tables for downscaling of CMIP6 datasets. I try to understand some of the attribute dependencies and i have some preliminary tables [here](https://github.com/WCRP-CORDEX/cordex-cmip6-cmor-tables/tree/main/Tables) and a [rendered example](https://wcrp-cordex.github.io/cordex-cmip6-cmor-tables/cmor-examples.html). For theses tests i have basically adapted the `CMIP6_CV` table and modified them. This all works fine, however, in CORDEX we don't require any `parent_*` or `sub_*` attributes and i tried to remove them completely from the CV and the dataset table. But somehow i can't get it to work, since i always get something like:

```
!!!!!!!!!!!!!!!!!!!!!!!!!
!
! Error: Your "sub_experiment_id" key could not be found in
! your Control Vocabulary file.
! 
!
!!!!!!!!!!!!!!!!!!!!!!!!!
```

So the cmorizer looks for that attribute although it is not defined or required either in the dataset table nor in the CV table. Is there some *special attributes* for CMOR3 that are required? I suspect so when i have a look at the source code, e.g., here
https://github.com/PCMDI/cmor/blob/c805fe0fcc509bf4ebed2024e193a009d791f9df/Src/cmor_CV.c#L821 it seems like this is somehow hardcoded and required?

At the same time, i wonder how this works, e.g., with the [input4MIps](https://github.com/PCMDI/input4MIPs-cmor-tables) or [obs4MIPs](https://github.com/PCMDI/obs4MIPs-cmor-tables) tables since they don't define, e.g., the `sub_experiment` attributes...

Maybe i get something totally wrong, i am not sure...

**Comment by larsbuntemeyer** [comment]
Ok, i now see that this behavious is activated by the presence of the `_cmip6_option` keyword in the dataset table... So i guess this should be absent if one wants to define an arbitrary CV?

**Comment by durack1** [comment]
@larsbuntemeyer you're almost there, if you wanted to see some of the `CMOR_input_example.json` inputs for the input4MIPs project, take a peek at [PCMDI/amipbcs/CMOR/drive_input4MIPs_obs.json](https://github.com/PCMDI/amipbcs/blob/master/CMOR/drive_input4MIPs_obs.json).

As you noted this does NOT include the `_cmip6_option` and thanks to the recent CMOR 3.7.0 release, depends on most of the inputs from the registered content (in the input4MIPs case information found in the https://github.com/PCMDI/input4MIPs-cmor-tables (and specifically the info found in the [PCMDI/input4MIPs-cmor-tables/Tables/input4MIPs_CV.json](https://github.com/PCMDI/input4MIPs-cmor-tables/blob/master/Tables/input4MIPs_CV.json)) file

**Comment by larsbuntemeyer** [comment]
> 

Yes, i didn't have the latest version, now this makes the input file really slim and convenient, very nice!!

**Comment by durack1** [comment]
@larsbuntemeyer it makes sense to me that we migrate this "issue" into a "discussion" as there is no obvious issue or bug that this thread identifies, rather it's about CMOR usage (and potential expansion/features)

**Comment by larsbuntemeyer** [comment]
I agree!
